File: Problems/Algorithms/38. Count and Say/CountSay.java
class Solution {
    public String countAndSay(int n) {
        if (n == 1) {
            return "1";
        }
        
        String prev = countAndSay(n-1);
        int i = 0; n = prev.length();
        String results = "";
        int count = 0;
        for (; i < n; i++) {
            if (i > 0 && prev.charAt(i) != prev.charAt(i-1)) {
                if (count > 0) {
                    results += count;
                    results += prev.charAt(i-1);
                }
                count = 1;
            } else {
                count++;
            }
        }
        results += count;
        results += prev.charAt(n-1);

        return results;
    }
}

File: Problems/Algorithms/38. Count and Say/count_say.py
#!/usr/bin/env python3
# coding: utf-8


class Solution:
    def countAndSay(self, n: int) -> str:
        if n == 1:
            return '1'
        else:
            s = self.countAndSay(n - 1)
            count = 1
            result = ''
            for i in range(1, len(s)):
                if s[i] == s[i - 1]:
                    count += 1
                else:
                    result += str(count) + s[i - 1]
                    count = 1
            result += str(count) + s[-1]
            return result
